### PR TITLE
Feature/audio

### PR DESCRIPTION
**Pull request by yalania**
This is not the hold feature, but we already added the libraries new clases, etc and I would like to merge to avoid having merging conflicts everytime.

Anyway the audio source can be used to do some testing if peple wants.

### CODE REVIEW

**Review by Pacasasgar — APPROVED** [review]
Some files from the Wwise project are still there, they should be removed. Rest is fine. Very good job!

File: Engine/Component/Component.h [hunk 1]
```diff
@@ -20,7 +20,8 @@ class Component
 		UI,
 		SCRIPT,
 		TRANSFORM2D,
-		ANIMATION
+		ANIMATION,
+		AUDIO_SOURCE
```
Comment (Dartemiss) [inline Engine/Component/Component.h]: Talk with @marcorod94 @vidalmelero @mariofv  to decide which component is component 9 and which is component 10 (audio/collider/UI)

File: Engine/Component/Component.h [hunk 2]
```diff
@@ -55,8 +56,17 @@ class Component
...
+	{
+		config.AddUInt(UUID, "UUID");
+		config.AddInt((unsigned int)type, "ComponentType");
+		config.AddBool(active, "Active");
+	};
+	virtual void Load(const Config &config)
+	{
+		UUID = config.GetUInt("UUID", 0);
+		active = config.GetBool("Active", true);
+	};
```
Comment (Dartemiss) [inline Engine/Component/Component.h:69]: @mariofv has done this on a different way in the UI branch. He will change it a bit on the merge.

File: Engine/EditorUI/Panel/InspectorSubpanel/PanelComponent.cpp [hunk 3]
```diff
@@ -24,6 +24,7 @@
 #include "Component/ComponentTransform.h"
 #include "Component/ComponentTransform2D.h"
 #include "Component/ComponentUI.h"
+#include "Component/ComponentAudioSource.h"
```
Comment (Dartemiss) [inline Engine/EditorUI/Panel/InspectorSubpanel/PanelComponent.cpp]: Codestyle

File: Engine/EditorUI/Panel/InspectorSubpanel/PanelComponent.h [hunk 4]
```diff
@@ -18,6 +18,7 @@ class ComponentUI;
 class ComponentText;
 class ComponentTransform;
 class ComponentScript;
+class ComponentAudioSource;
 class GameObject;
```
Comment (Dartemiss) [inline Engine/EditorUI/Panel/InspectorSubpanel/PanelComponent.h]: Code style

File: Engine/Module/ModuleScriptManager.cpp [hunk 5]
```diff
@@ -210,7 +210,10 @@ void ModuleScriptManager::ReLink()
 {
 	for (const auto& component_script : scripts)
 	{
-		component_script->script->Link();
+		if (component_script->script)
+		{
+			component_script->script->Link();
+		}
```
Comment (Dartemiss) [inline Engine/Module/ModuleScriptManager.cpp:216]: What is this if for?

**Review by Dartemiss — APPROVED** [review]
Good job guys! Just a few comments that would be nice to change. Also very important to talk with the other teams for the numeration of the new components for not destroying scenes.

File: Engine/Module/ModuleScriptManager.cpp (hunk 5 above)
Comment (yalania) [reply Engine/Module/ModuleScriptManager.cpp:216]: in case the script is nullprt this will crash so I just controlled it

**Pull request merged**